diag_manager/diag_data.F90

```diff
@@ -86,6 +108,33 @@ MODULE diag_data_mod
 
   !> @}
 
+
+  !> @brief The files type matching a C struct containing diag_yaml information
+    !> @ingroup diag_data_mod
+type, bind(c) :: diag_files_type
```

Comment:
Why is this in c?

Reply:
Why shouldn't it be in C?  We can get this info from the parser in C and it will transfer over to Fortran for free.  There is still other information we will need to use the Fortran parser for.  It's copied over from this struct to the Fortran object.

Would you prefer it handled in Fortran?  None of the actual parsing is coded yet, so this is still open for discussion.

Reply:
> Why shouldn't it be in C? We can get this info from the parser in C and it will transfer over to Fortran for free.

How are you getting this info from the parser in c? Are you going to be #rollingyourown code that calls libyaml directly? 

> Would you prefer it handled in Fortran? 

We already have something that parsers a yaml file in fortran, why would we not use it?

Reply:
OK, I can update this. 

> > Why shouldn't it be in C? We can get this info from the parser in C and it will transfer over to Fortran for free.
> 
> How are you getting this info from the parser in c? Are you going to be #rollingyourown code that calls libyaml directly?
It wouldn't be #rollyourown if we are calling the parser directly.  It would be using the parser.
> 
> > Would you prefer it handled in Fortran?
> 
> We already have something that parsers a yaml file in fortran, why would we not use it?
We will have to use the Fortran side of it no matter what because not everything would be in this struct.  The struct isn't coded yet and neither are any of the parser calls, so there's no reason why I can't remove this and do everything on the Fortran side.